ecommerce_extensions/payment/views/ednx_pp.py

```diff
@@ -0,0 +1,209 @@
+    @method_decorator(transaction.non_atomic_requests)
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super().dispatch(request, *args, **kwargs)
+
+    def _get_billing_address(self, ednx_response):
+
+        return BillingAddress(
+            first_name=ednx_response.get('cc_holder', ''),
+            last_name=ednx_response.get('cc_holder', ''),
+            line1=ednx_response.get('billing_address', ''),
+            line4=ednx_response.get('billing_city', ''),  # Oscar uses line4 for city
+            country=Country.objects.get(iso_3166_1_a2=ednx_response['billing_country']),
+        )
+
+    def _get_basket(self, basket_id):
+        if not basket_id:
+            return None
+
+        try:
+            basket_id = int(basket_id)
+            basket = Basket.objects.get(id=basket_id)
+            basket.strategy = strategy.Default()
+            Applicator().apply(basket, basket.owner, self.request)
+            return basket
+        except (ValueError, ObjectDoesNotExist):
+            return None
+
+    def post(self, request):
+        """Process a PP merchant notification and place an order for paid products as appropriate."""
+
+        # Note (CCB): Orders should not be created until the payment processor has validated the response's signature.
+        # This validation is performed in the handle_payment method. After that method succeeds, the response can be
+        # safely assumed to have originated from ednx.
+        ednx_response = request.POST.dict()
+        basket = None
+        transaction_id = None
+
+        try:
+            transaction_id = ednx_response.get('transaction_id')
+            order_number = ednx_response.get('reference_sale')
+            basket_id = OrderNumberGenerator().basket_id(order_number)
+
+            logger.info(
+                'Received PP merchant notification for transaction [%s], associated with basket [%d].',
+                transaction_id,
+                basket_id
+            )
+
+            basket = self._get_basket(basket_id)
+
+            if not basket:
+                logger.error('Received payment for non-existent basket [%s].', basket_id)
+                return HttpResponse(status=status.HTTP_400_BAD_REQUEST)
+        finally:
+            # Store the response in the database.
+            ppr = self.payment_processor.record_processor_response(
+                ednx_response,
+                transaction_id=transaction_id,
+                basket=basket,
+            )
+
+        try:
+            # Explicitly delimit operations which will be rolled back if an exception occurs.
+            with transaction.atomic():
+                try:
+                    self.handle_payment(ednx_response, basket)
+                except InvalidSignatureError:
+                    logger.exception(
+                        'Received an invalid ednx response. The payment response was recorded in entry [%d].',
```

Comment:
I'm not convinced about the `ednx response`, maybe Payment Processor response, or what this is